### PR TITLE
 Migrate stable diffusion & controlnet components to PandasTransformComponent

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Contributes to #203 

This PR updates the reusable components of the stable diffusion / controlnet pipeline to use the new Pandas interface. I submit these separately since I've already tested them. 

Some simplifications are clear:
- No more `map_partitions` needed for batching
- No more `meta` needed for `apply` or similar methods

Since we now set the `meta` of the `map_partitions` in the `PandasTransformComponent` based on the component spec, the returned dataframes are validated quite strictly. It does not allow any additional columns.

The Pandas implementation can probably be improved further as I'm not the biggest Pandas pro.

### CODE REVIEW

File: components/caption_images/src/main.py [hunk 1]
```diff
@@ -53,69 +52,39 @@ def caption_image_batch(
...
-            batch_size=batch_size,
-            max_new_tokens=max_new_tokens,
-            device=device,
-            meta={0: str}
+    def setup(self, *, model_id: str, batch_size: int, max_new_tokens: int) -> None:
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        logger.info(f"Device: {self.device}")
+
+        self.processor = BlipProcessor.from_pretrained(model_id)
+        self.model = BlipForConditionalGeneration.from_pretrained(model_id)
```
Comment (NielsRogge) [inline components/caption_images/src/main.py:63]: It doesn't seem like the model is put on the `device`

File: components/segment_images/src/main.py [hunk 2]
```diff
@@ -76,64 +75,42 @@ def segment_image_batch(image_batch: pd.DataFrame, *, model: AutoModelForSemanti
...
-            model=model,
-            processor=processor,
-            batch_size=batch_size,
-            device=device,
-            meta={0: object}
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        logger.info(f"Device: {self.device}")
+
+        self.processor = SegformerImageProcessor.from_pretrained(model_id)
+        self.model = AutoModelForSemanticSegmentation.from_pretrained(model_id)
```
Comment (NielsRogge) [inline components/segment_images/src/main.py:91]: Same here, model needs to be put on the `device`

**Review by NielsRogge — APPROVED** [review]
Looks very nice!

File: components/caption_images/src/main.py (hunk 1 above)
Comment (RobbeSneyders) [reply components/caption_images/src/main.py]: Good catch. Seems like this wasn't done before either.

**Pull request merged**